PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/datalink.cpp

```diff
@@ -1,6 +1,115 @@
 #include "datalink.h"
 
-Flit *DatalinkLayer::addDLLP(Flit *flit, Dllp::DllpType dllpType, Dllp::CreditType creditType, bool shared, int credit[])
+/**
+ * @brief Updates the shared credit limit values based on the received DLLP information
+ * @param flit Flit containing the DLLPs
+ * @param P_SHARED_CREDIT_LIMIT Array of shared credit limit for posted traffic
+ * @param NP_SHARED_CREDIT_LIMIT Array of shared credit limit for non-posted traffic
+ * @param CPL_SHARED_CREDIT_LIMIT Array of shared credit limit for completion traffic
+ * @param FI1 First indication (FI1) flag for credit update
+ * @param FI2 Second indication (FI2) flag for credit update
+*/
+void DatalinkLayer::updateCreditLimit(Flit flit, int P_SHARED_CREDIT_LIMIT[], int NP_SHARED_CREDIT_LIMIT[], int CPL_SHARED_CREDIT_LIMIT[], int P_DEDICATED_CREDIT_LIMIT[], int NP_DEDICATED_CREDIT_LIMIT[], int CPL_DEDICATED_CREDIT_LIMIT[], bool& FI1, bool& FI2) {
+	// Static variables to store the flags and state across function calls all initialized to false
+	static bool discard, sharedPFC2Flag, sharedNPFC2Flag, sharedCPLFC2Flag, dedicatedPFC2Flag, dedicatedNPFC2Flag, dedicatedCPLFC2Flag;
```

Comment:
I believe you don't need any of these flags anymore since the logic is to set FI2 after FI1 is set and you receive any initFC2 DLLP. Checking if all is not -1 is enough to set FI1, and from here just discard all FC1 DLLPs and set FI2 on receiving of any FC2 DLLP

Reply:
What I mean here is to represent it as a state machine: case if FI1 is not set and case if it is set